### DISCUSSION TITLE
How can I get the bot to answer in other language other than English?

### DISCUSSION THREAD

**Discussion by Jackelope007** [FlowiseAI/Flowise#453] [General]
I have put together this chatbot for my page and fed it a ton of documents that are in German. And I ask the bot questions in German but it continuously answers back in English. The answers are right which is GREAT, but the language part in driving me a bit crazy. I change the original prompt to German thinking that might help but nope. 
Any ideas?
Thx in advance

![Screenshot 2023-06-30 075900](https://github.com/FlowiseAI/Flowise/assets/127890539/1a3c83ea-a374-4c1d-bbe4-a57a73d5b6e9)
![Screenshot 2023-06-30 075915](https://github.com/FlowiseAI/Flowise/assets/127890539/7a30ea38-6f93-46fe-aa44-b033ac465139)

**Comment by HenryHengZJ** [comment]
This requires a little bit trial and error. Try this: 
```
I want you to act as a document that I am having a conversation with. Your name is "AI Bot". You will provide me with answers from the given info. Only reply me in German. If the answer is not included, say exactly "Ich weiß nicht" and stop after that. Refuse to answer any question not about the info. Never break character.
```

**Reply by Jackelope007** [reply]
thank you. I tried that and its a hit or miss. Sometimes it answers in german and sometimes in english.

**Comment by joaquin-torroba** [comment]
It doesn't work if you put in the prompt system as a "mandatory condition" to always answer in german? I do that in Spanish and is working.

**Reply by Jackelope007** [reply]
Ok, so I should write in the prompt "Mandatory condition" with the language request? Spanish is actually another language I want to add but that will be after I get the english / german figured out.

**Reply by joaquin-torroba** [reply]
For me is working. 
Here a quick example of one chatflow for ads copywriting (see the yellow part):

![image](https://github.com/FlowiseAI/Flowise/assets/137190362/bebbbf1b-31fa-46ef-9039-be5675aee498)

**Reply by Jackelope007** [reply]
oh damn. Ok, I see now the detail to put into it. I will try it like that. Thank you for taking the time. I'll post if it worked so hopefully it'll help others.

**Reply by joaquin-torroba** [reply]
Yes but I just tried with a simpler one and got the same results. 
First the answer was in english, so I added one line at the end of the system prompt ("ALL your answers should be in spanish" - see image 1) and tested it and it replied in spanish (see image 2).

![image](https://github.com/FlowiseAI/Flowise/assets/137190362/11d6064d-7ff6-4853-8e5f-bd756355fa58)

![image](https://github.com/FlowiseAI/Flowise/assets/137190362/c3ea6d24-34ab-4811-85e9-ead71f86adc8)

**Reply by HenryHengZJ** [reply]
If you sync your fork to latest changes, I've modified the prompt to let the chain answer in its original language - see [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/chains/ConversationalRetrievalQAChain/ConversationalRetrievalQAChain.ts#L9)

I can see more stable performance now. If you are using npx or docker, its still not yet pushed to new release

**Comment by DevOtts** [comment]
@HenryHengZJ one of the problem is that the default prompts are in English. I would like to contribute at least to improving the `ConversationalRetrievalQAChain_Chains` experience.
What I'm doing here is to add an additional Parameter called "Internal Prompt Language" where I'll let the user choose which language he wants. In the code I'm creating more variants of `/prompts.ts` like `language/en_us/prompts.ts`, `language/pt_br/prompts.ts`,`language/es_es/prompts.ts`, and so on using the [scripts language codes](https://www.softaculous.com/docs/admin/scripts-language-codes/) as the pattern.

Am I able to create a PR with that change?

<img width="700" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/851664/65c40da9-66e1-41fc-8c96-9a32ee98a148">
